### PR TITLE
[s] Adds a one in a million chance to win a pulse rifle from arcade machines

### PR DESCRIPTION
**Pull request by Incoming5643**
It is 4:30 AM and the darkness is overtaking my mind

:cow2:

### CODE REVIEW

**Comment by TechnoAlchemisto** [comment]
This is good I like this, I just spent the last hour recovering my car from a towing yard because I parked my car in a korean market parking lot because I wanted to go see akira in the theater across from it but they towed my car those korean sons of bitches 

so yeah now i'm a straight 400 dollars poorer and drunk, hope your pr gets merged :woohoo:

**Comment by xxalpha** [comment]
One in a million

Or

A million in one

Byond RNG can go either way.

**Comment by Incoming5643** [comment]
PR doubles as an easy way to tell if byond ever breaks prob()

**Comment by Giacom** [comment]
Wait, AND they get their original prize? I think that is too OP!

**Comment by Incoming5643** [comment]
They don't still get their original prize, because it runs a check to see if there's any contents in the arcade to award first.

Yeah there's a feature we've had I never knew: Putting things in arcade cabinet's contents to give out custom prizes.

**Comment by Giacom** [comment]
Oh cool

**Comment by bgobandit** [comment]
what could possibly go wrong

**Comment by xxalpha** [comment]
I suggest logging when a pulse rifle is awarded.

File: code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -50,6 +50,9 @@
 		qdel(src)
 
 /obj/machinery/computer/arcade/proc/prizevend()
+	if(prob(prob(prob(1)))) //1 in a million
+		new /obj/item/weapon/gun/energy/pulse(contents)
+
```
Comment (nullbear) [inline code/game/machinery/computer/arcade.dm]: im thinking what people are saying, is to add a return here.

**Comment by octareenroon91** [comment]
Would you consider switching to anyprob() for readability?

File: code/game/machinery/computer/arcade.dm (hunk 1 above)
Comment (Giacom) [inline code/game/machinery/computer/arcade.dm]: Actually I was wrong, adding a return will just spawn the gun in the arcade's contents.

Letting it continue will then dispense it as normal.

File: code/game/machinery/computer/arcade.dm [hunk 2]
```diff
@@ -50,6 +50,9 @@
 		qdel(src)
 
 /obj/machinery/computer/arcade/proc/prizevend()
+	if(prob(prob(prob(1)))) //1 in a million
```
Comment (MrStonedOne) [inline code/game/machinery/computer/arcade.dm]: could just do 000.1

**Comment by MrStonedOne** [comment]
This is the kind of feature that will get forgotten until somebody wins and everybody will go WHY WAS THAT CODE IN THERE!

**Comment by Razharas** [comment]
But that reaction is the sole purpose of this pr

**Comment by ShadowDeath6** [comment]
Should be pulse pistol xdd

**Comment by octareenroon91** [comment]
1911-P then.

**Comment by Bawhoppen** [comment]
I vote for multiverse sword instead it fits arcade more

**Comment by PKPenguin321** [comment]
^

**Comment by Durkel** [comment]
Should add a pinball machine and give it a one in a million chance to get a spellbook.

**Comment by octareenroon91** [comment]
Jesus guys how long will it take for a million arcade rounds to even be attempted?

**Comment by kilkun** [comment]
Energy Axe would be a better prize.

**Comment by Cheridan** [comment]
>I suggest logging when a pulse rifle is awarded.

It should send a fanfare and a server-wide message saying just how lucky this spaceman is so that everyone can congratulate him.

**Comment by Jazaen** [comment]
This probably will never, ever happen. Barring BYOND RNG, of course. If we had 50 spessmen, winning an arcade game each minute, during entire 60-minute round, it would take 334 such rounds to statistically net one pulse rifle, and said winner would probably be torn to shreds by everyone else.

Merge it.

**Comment by Alek2ander** [comment]
> Scientists have calculated that the chances of something so patently absurd actually existing are millions to one.
But magicians have calculated that million-to-one chances crop up nine times out of ten.

**Comment by Iamgoofball** [comment]
watch as 5 people win it within the first day

**Comment by Niknakflak** [comment]
Watch as a bug makes it so everytime you win an arcade it spawns a pulse rifle

**Comment by Zilenan91** [comment]
>watch as it's intentionally coded so that you """""""accidentally""""""" win a pulse rifle every time until it's fixed a day later

File: code/game/machinery/computer/arcade.dm (hunk 2 above)
Comment (monster860) [inline code/game/machinery/computer/arcade.dm]: That's 1 in a thousand. I'm pretty sure byond doesn't work that way too

File: code/game/machinery/computer/arcade.dm [hunk 3]
```diff
@@ -50,6 +50,9 @@
...
+		new /obj/item/weapon/gun/energy/pulse(contents)
```
Comment (RemieRichards) [inline code/game/machinery/computer/arcade.dm]: ```contents``` is not a loc, lists are not locs, this just gives me -weird- vibes, swap it for ```(src)```

**Comment by Anonmare** [comment]
Discworld has taught me that one in a million chances happen nine times out of ten

**Comment by PKPenguin321** [comment]
We need a 1 in a million chance for disarm attacks to violently sever the arms of the victim

File: code/game/machinery/computer/arcade.dm (hunk 2 above)
Comment (Incoming5643) [inline code/game/machinery/computer/arcade.dm]: prob(100) 1 in 1
prob(10) 1 in 10
prob(1) 1 in 100
prob(0.1) 1 in 1000
prob(0.01) 1 in 10,000
prob(0.001) 1 in 100,000
prob(0.0001) 1 in 1,000,000

Rough decimal placement not withstanding

**Comment by Incoming5643** [comment]
Now we wait

**Pull request merged**